pkg/api/bzz.go

```diff
@@ -96,24 +103,103 @@ func (s *server) bzzDownloadHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
-	me, err := m.Lookup(path)
+	if pathVar == "" {
+		logger.Tracef("bzz download: handle empty path %s", address)
+
+		indexDocumentSuffixKey := bzzDownloadHandlerManifestRedirect(m, manifestWebsiteIndexDocumentSuffixKey)
+		if indexDocumentSuffixKey != "" {
+			pathWithIndex := path.Join(pathVar, indexDocumentSuffixKey)
+			indexDocumentManifestEntry, err := m.Lookup(pathWithIndex)
+			if err == nil {
+				// index document exists
+				logger.Debugf("bzz download: serving path: %s", pathWithIndex)
+
+				s.bzzDownloadHandlerServeManifestEntry(w, r, ctx, j, address, indexDocumentManifestEntry.Reference())
+				return
+			}
+		}
+	}
+
+	me, err := m.Lookup(pathVar)
 	if err != nil {
-		logger.Debugf("bzz download: invalid path %s/%s: %v", address, path, err)
+		logger.Debugf("bzz download: invalid path %s/%s: %v", address, pathVar, err)
 		logger.Error("bzz download: invalid path")
 
 		if errors.Is(err, manifest.ErrNotFound) {
+
+			if !strings.HasPrefix(pathVar, "/") {
+				// check for directory
+				dirPath := pathVar + "/"
+				exists, err := m.HasPrefix(dirPath)
+				if err == nil && exists {
+					// redirect to directory
+					u := r.URL
+					u.Path += "/"
+					redirectURL := u.String()
+
+					logger.Debugf("bzz download: redirecting to %s: %v", redirectURL, err)
+
+					http.Redirect(w, r, redirectURL, http.StatusPermanentRedirect)
+					return
+				}
+			}
+
+			// check index suffix path
+			indexDocumentSuffixKey := bzzDownloadHandlerManifestRedirect(m, manifestWebsiteIndexDocumentSuffixKey)
+			if indexDocumentSuffixKey != "" {
+				if !strings.HasSuffix(pathVar, indexDocumentSuffixKey) {
+					// check if path is directory with index
+					pathWithIndex := path.Join(pathVar, indexDocumentSuffixKey)
+					indexDocumentManifestEntry, err := m.Lookup(pathWithIndex)
+					if err == nil {
+						// index document exists
+						logger.Debugf("bzz download: serving path: %s", pathWithIndex)
+
+						s.bzzDownloadHandlerServeManifestEntry(w, r, ctx, j, address, indexDocumentManifestEntry.Reference())
+						return
+					}
+				}
+			}
+
+			// check if error document is to be shown
+			errorDocumentPath := bzzDownloadHandlerManifestRedirect(m, manifestWebsiteErrorDocumentPathKey)
+			if errorDocumentPath != "" {
+				if pathVar != errorDocumentPath {
+					errorDocumentManifestEntry, err := m.Lookup(errorDocumentPath)
+					if err == nil {
+						// error document exists
+						logger.Debugf("bzz download: serving path: %s", errorDocumentPath)
+
+						s.bzzDownloadHandlerServeManifestEntry(w, r, ctx, j, address, errorDocumentManifestEntry.Reference())
+						return
+					}
+				}
+			}
+
 			jsonhttp.NotFound(w, "path address not found")
 		} else {
 			jsonhttp.BadRequest(w, "invalid path address")
 		}
 		return
 	}
 
-	manifestEntryAddress := me.Reference()
+	// serve requested path
+	s.bzzDownloadHandlerServeManifestEntry(w, r, ctx, j, address, me.Reference())
+}
+
+func (s *server) bzzDownloadHandlerServeManifestEntry(
```

Comment:
maybe just `func serveManifestEntry` is enough? not sure if we need too much nesting in function naming too